app/src/main/java/org/fossasia/susi/ai/activities/MainActivity.java

```diff
@@ -981,7 +980,7 @@ private void checkPermissions() {
 
         if(ActivityCompat.checkSelfPermission(this,
                 Manifest.permission.RECORD_AUDIO) == PackageManager.PERMISSION_GRANTED) {
-            PrefManager.putBoolean(Constant.MIC_INPUT, checkMicInput());
+            PrefManager.putBoolean(Constant.MIC_INPUT, checkMicInput() && PrefManager.getBoolean(Constant.MIC_INPUT, false));
```

Comment:
Avoid changes in the main activity, please. The main activity will be removed at all and these changes won't reflect then. Instead, comment these changes with the line number in #832. I'll add them.